crates/ruff_db/src/diagnostic/mod.rs

```diff
@@ -249,8 +249,8 @@ impl SubDiagnostic {
     ///
     /// If a sub-diagnostic has not been marked as printed before being
     /// dropped, then its `Drop` implementation will panic in debug mode.
-    #[cfg(debug_assertions)]
-    pub(crate) fn printed(&mut self) {
+    pub fn printed(&mut self) {
```

Comment:
@BurntSushi I hope you're fine with me making this method public. I need a way to say: Yes, I printed this diagnostic, please don't panic ;)

https://github.com/astral-sh/ruff/blob/54c359b8fc5cc65e41d73f5a4573b5b1a93ca4b0/crates/red_knot_ide/src/goto.rs#L857-L884

Reply:
Oh I see. OK, I actually looked at how you're using this and see what you mean.

I think for this particular example, one thing that would be nice is to define a `Diagnostic::ignore`/`SubDiagnostic::ignore` method that consumes `self` and calls the crate internal `printed()` method. That way, it's a declarative and compiler enforced way of "throwing away" a diagnostic intentionally.

It probably doesn't cover everything that the more flexible `printed()` method does, but I think it's probably nicer if we can get away with it.

Reply:
I rewrote it so that the `source` gets recreated everytime (by having a specific `GotoTypeDefinitionDiagnostic`) but I also like your idea of having an `ignore` method which "explicitly" drops a diagnostic (which will be something we need for Ruff where `Diagnostic`s get created even if they're suppressed and are then filtered out at a later stage)